src/Controller/Adminhtml/Product/CreateProjectFromProduct.php

```diff
@@ -0,0 +1,125 @@
+use EasyTranslate\Connector\Model\ProjectFactory;
+use EasyTranslate\RestApiClient\Workflow;
+use Exception;
+use Magento\Backend\App\Action;
+use Magento\Backend\App\Action\Context;
+use Magento\Framework\Api\DataObjectHelper;
+use Magento\Framework\App\Request\DataPersistorInterface;
+use Magento\Framework\Exception\LocalizedException;
+use Magento\Ui\Component\MassAction\Filter;
+
+class CreateProjectFromProduct extends Action
+{
+    /**
+     * @var Team
+     */
+    private $team;
+
+    /**
+     * @var ProjectFactory
+     */
+    private $projectFactory;
+
+    /**
+     * @var DataObjectHelper
+     */
+    private $dataObjectHelper;
+
+    /**
+     * @var ProjectDataProcessor
+     */
+    private $projectDataProcessor;
+
+    /**
+     * @var DataPersistorInterface
+     */
+    private $dataPersistor;
+
+    public function __construct(
+        Context $context,
+        Team $team,
+        DataPersistorInterface $dataPersistor,
+        DataObjectHelper $dataObjectHelper,
+        ProjectDataProcessor $projectDataProcessor,
+        ProjectFactory $projectFactory
+    ) {
+        parent::__construct($context);
+        $this->team                 = $team;
+        $this->projectFactory       = $projectFactory;
+        $this->dataObjectHelper     = $dataObjectHelper;
+        $this->projectDataProcessor = $projectDataProcessor;
+        $this->dataPersistor        = $dataPersistor;
+    }
+
+    public function execute()
+    {
+        $resultRedirect = $this->resultRedirectFactory->create();
+        try {
+            $data = $this->projectData();
+        } catch (Exception $exception) {
+            return $resultRedirect->setPath('catalog/product/edit/id/' . $this->getProductIds()[0]);
+        }
+        $project = $this->projectFactory->create();
+        $this->dataObjectHelper->populateWithArray($project, $data, ProjectInterface::class);
+        $project = $this->projectDataProcessor->saveProjectPostData($project, $data);
+        $this->dataPersistor->clear('easytranslate_project');
+        $this->messageManager->addSuccessMessage((string)__('You have created a new project'));
+
+        return $resultRedirect->setPath(
+            'easytranslate/project/edit',
+            [ProjectInterface::PROJECT_ID => $project->getProjectId()]
+        );
+    }
+
+    /**
+     * @throws LocalizedException
+     */
+    private function projectData(): array
+    {
+        return [
+            ProjectInterface::PRODUCTS         => $this->getProductIds(),
+            ProjectInterface::NAME             => 'Easytranslate Project Name',
+            ProjectInterface::TEAM             => $this->getTeam(),
+            ProjectInterface::SOURCE_STORE_ID  => 0,
+            ProjectInterface::STATUS           => Status::OPEN,
+            ProjectInterface::PRICE            => null,
+            ProjectInterface::WORKFLOW         => Workflow::TYPE_TRANSLATION,
+            ProjectInterface::TARGET_STORE_IDS => [1],
```

Comment:
Actually we dont, we can just pass in an empty array. I did this initially because when manually creating a project the target store ids are a required field.